core_lib/src/interface/timecontrols.cpp

```diff
@@ -291,10 +340,74 @@ void TimeControls::updateSoundScrubIcon(bool soundScrubEnabled)
     }
 }
 
+void TimeControls::noTimecodeText()
+{
+    QSettings settings(PENCIL2D, PENCIL2D);
+    settings.setValue(SETTING_TIMECODE_TEXT, NOTEXT);
+    mTimecodeLabelEnum = NOTEXT;
+    mTimecodeLabel->setToolTip(tr(""));
+    updateTimecodeLabel(mEditor->currentFrame());
+}
+
+void TimeControls::onlyFramesText()
+{
+    QSettings settings(PENCIL2D, PENCIL2D);
+    settings.setValue(SETTING_TIMECODE_TEXT, FRAMES);
+    mTimecodeLabelEnum = FRAMES;
+    mTimecodeLabel->setToolTip(tr("Actual frame number"));
+    updateTimecodeLabel(mEditor->currentFrame());
+}
+
+void TimeControls::SFF_text()
+{
+    QSettings settings(PENCIL2D, PENCIL2D);
+    settings.setValue(SETTING_TIMECODE_TEXT, SFF);
+    mTimecodeLabelEnum = SFF;
+    mTimecodeLabel->setToolTip(tr("Timecode format S:FF"));
+    updateTimecodeLabel(mEditor->currentFrame());
+}
+
+void TimeControls::SMPTE_text()
+{
+    QSettings settings(PENCIL2D, PENCIL2D);
+    settings.setValue(SETTING_TIMECODE_TEXT, SMPTE);
+    mTimecodeLabelEnum = SMPTE;
+    mTimecodeLabel->setToolTip(tr("Timecode format MM:SS:FF"));
+    updateTimecodeLabel(mEditor->currentFrame());
+}
+
 void TimeControls::onFpsEditingFinished()
 {
     mFpsBox->clearFocus();
     emit fpsChanged(mFpsBox->value());
+    mFps = mFpsBox->value();
+}
+
+void TimeControls::updateTimecodeLabel(int frame)
+{
+    switch (mTimecodeLabelEnum)
+    {
+    case TimecodeTextLevel::SMPTE:
+        mTimecodeLabel->setText(QString("%1:%2:%3")
+                                .arg(QString::number(frame / (60 * mFps) % 60).rightJustified(2,'0'))
+                                .arg(QString::number(frame / mFps % 60).rightJustified(2, '0'))
+                                .arg(QString::number(frame % mFps).rightJustified(2, '0')));
+        break;
+    case TimecodeTextLevel::SFF:
+        mTimecodeLabel->setText(QString("%1:%2")
+                                .arg(QString::number(frame / mFps))
+                                .arg(QString::number(frame % mFps).rightJustified(2, '0')));
+        break;
+    case TimecodeTextLevel::FRAMES:
+        mTimecodeLabel->setText(tr("%1").arg(QString::number(frame).rightJustified(4, '0')));
+        break;
+    case TimecodeTextLevel::NOTEXT:
+    default:
+        mTimecodeLabel->setText("");
```

Comment:
What can NOTEXT be, if it's not an empty string?

Reply:
Sorry, I should have been more explicit. What I (failed to) imply was that the label should be completely hidden from the toolbar when NOTEXT is selected, so that it doesn’t consume any space.

Reply:
Done, but it's hard to see that it gives more space to the right.

Reply:
You mean it’s hard to see that something else can be made to appear there?

Personally, I don’t find that empty space to be more meaningful than no space at all. Actually, that empty space looks kinda weird to me, which is why I requested this change.

Keep in mind that the default is to show the frame number. That means that if the timecode format has been set to “No Text”, the user *chose* to hide that part of the UI. Moreover, since they hid it, they already know that this option is available. That’s my take on the matter at least.

This aside, hiding the label doesn’t actually seem to work right now. That said, I don’t see any issues with the code either. I’ll try to figure out what’s going on there.

Reply:
Figured it out. `addWidget(mTimecodeLabel)` returns a QAction which must be used to control the visibility.

> **Note:** You should use QAction::setVisible() to change the visibility of the widget. Using QWidget::setVisible(), QWidget::show() and QWidget::hide() does not work.

Reply:
I've tried what you said @J5lx . Here's the code:
```
    case TimecodeTextLevel::NOTEXT:
    default:
        mNoTimecodeAction->setVisible(false);
        mTimecodeLabel->setVisible(false);
        break;
```
...and it doesn't work. It is mNoTimecodeAction you're referring to? 
I've tried both with and without the " mTimecodeLabel->setVisible(false);".
I'm sorry, but there must be something I misunderstand.
Is the easiest way out, that you do the necessary changes, and I merge them?

Reply:
When you call `addWidget(mTimecodeLabel)` you get a QAction as return value. Right now, that return value is ignored because it isn’t needed for anything else. But it is the action that setVisible() needs to be called on. I’ll send a PR in a moment.

Reply:
Check davidlamhauge#16